lib/recursive_open_struct.rb

```diff
@@ -10,16 +10,21 @@ class RecursiveOpenStruct < OpenStruct
   include DebugInspect
 
   def initialize(hash=nil, args={})
-    hash ||= {}
-    @recurse_over_arrays = args.fetch(:recurse_over_arrays, false)
-    @deep_dup = DeepDup.new(recurse_over_arrays: @recurse_over_arrays)
+    @args = args
+    @table = hash || {}
+    init
+  end
 
-    @table = args.fetch(:mutate_input_hash, false) ? hash : @deep_dup.call(hash)
-    @table && @table.each_key { |k| new_ostruct_member(k) }
+  def init
```

Comment:
This method should be marked `protected` or `private`. It should only used internally by ROS.